lib/msal-angular/README.md

```diff
@@ -2,49 +2,125 @@
+    ]
+   },
+  { path: 'myProfile' ,component: MsGraphComponent, canActivate : [MsalGuard] },
+```
 
-protectedResourceMap.set ("https://graph.microsoft.com/v1.0/me", ["user.read"]);
-protectedResourceMap.set("https://buildtodoservice.azurewebsites.net/api/todolist", ["api://a88bb933-319c-41b5-9f04-eff36d985612/access_as_user"]);
+When user visits these routes, the library prompts the user to authenticate.
 
-export function loggerCallback(logLevel, message, piiEnabled) {
-  
-}
+#### 3. Get tokens for API calls
+MSAL Angular allows you to add an Http interceptor (`MsalInterceptor`) in your app.module.ts as follows. MsalInterceptor will obtain tokens and add them to all your Http requests in API calls except the API endpoints listed as `unprotectedResources`.
+
+```js
+providers: [ ProductService, {
+        provide: HTTP_INTERCEPTORS,
+        useClass: MsalInterceptor,
+        multi: true
+    }
+   ],
+ ```
+
+Using MsalInterceptor is optional and you can write your own interceptor if you choose to. Alternatively, you can also explicitly acquire tokens using the acquireToken APIs.
+
+#### 4. Subscribe to event callbacks
+
+MSAL wrapper provides below callbacks for various operations. For all callbacks, you need to inject BroadcastService as a dependency in your component/service.
+
+1. login_popup()/login_redirect using api or using routes.
+
+```js
+this.broadcastService.subscribe("msal:loginFailure", (payload) => {
+// do something here
+});
+
+this.broadcastService.subscribe("msal:loginSuccess", (payload) => {
+// do something here
+});
+```
+
+2. acquire_token_silent()/acquire_token_popup()/acquire_token_redirect()
 
+```js
+this.broadcastService.subscribe("msal:acquireTokenSuccess", (payload) => {
+     // do something here
+});
+
+this.broadcastService.subscribe("msal:acquireTokenFailure", (payload) => {
+      // do something here
+});
+```
+
+3. It is extremely important to unsubscribe. Implement ngOnDestroy() in your component and unsubscribe.
+
+```js
+ private subscription: Subscription;
+
+ this.subscription=  this.broadcastService.subscribe("msal:acquireTokenFailure", (payload) => {
+ }
+
+ ngOnDestroy() {
+    this.broadcastService.getMSALSubject().next(1);
+    if(this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+```
+
+## Samples
+
+You can find a quickstart and detailed sample under the [sample directory](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-angular/samples).
+
+## MSAL Angular public API
+
+#### Config options for MSAL properties
+
+Besides the required clientID, you can optionally pass the following config options to MSAL module during initialization. For example:
+
+```js
 @NgModule({
   imports: [ MsalModule.forRoot({
                   clientID: Your client ID,
-                  authority: "https://login.microsoftonline.com/microsoft.onmicrosoft.com/",
+                  authority: "https://login.microsoftonline.com/contoso.onmicrosoft.com/",
```

Comment:
why contoso? Is this a standard practice? Our samples are still using microsoft.onmicrosoft.com